lib/page/surveydetail/survey_detail_page.dart

```diff
@@ -66,16 +141,37 @@ class SurveyDetailPage extends StatelessWidget {
                       buttonText:
                           AppLocalizations.of(context)!.surveyDetailStartSurvey,
                       onPressed: () {
-                        // TODO: Navigate to Questions screen
+                        if (surveyDetail != null) {
+                          _navigateToQuestion(surveyDetail);
+                        } else {
+                          if (survey != null)
+                            ref
+                                .read(surveyDetailViewModelProvider.notifier)
+                                .getSurveyDetail(survey, isRetry: true);
+                        }
```

Comment:
So, in case of an error, clicking on this Next button will retry the API request and navigate to the next screen. Does this logic come from the IC's requirement? Do we need to apply this retry logic in all API requests in the whole app or only here?

Sorry but I'm quite curious why do you add this logic here? 🤔

Reply:
@luongvo FYI I just found a bug when retrying getSurveyDetail success. 🙈  The surveyDetail from the stream is always null so it won't navigate to the Questions screen. Fixed in https://github.com/chornerman/Survey/pull/90/commits/8edc612f4a7577e17418a38c7e11b10f3d4cfbb4

Reply:
@chornerman Just my thought, it's an unnecessary feature. Since you already implemented it, I'll review it by agreeing to have that feature in your IC.